src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -727,9 +731,11 @@ private NamedTypeSymbol ApplyGenericArguments(NamedTypeSymbol symbol, Type[] typ
             AssemblyIdentity assemblyOpt,
             bool includeReferences,
             bool isWellKnownType,
+            out ImmutableArray<AssemblySymbol> conflicts,
             DiagnosticBag warnings = null,
             bool ignoreCorLibraryDuplicatedTypes = false)
         {
+            conflicts = default;
```

Comment:
>conflicts = default; [](start = 12, length = 20)

We usually prefer an empty array in cases like this. Also, does ```conflicts``` have to be an array? It feels like this is an allocation that we can avoid. Can it be just a tuple?

Reply:
Changed to tuple. However, I'm not sure the allocation was critical here, because it's an error scenario.

---
In reply to: [146343923](https://github.com/dotnet/roslyn/pull/22793#discussion_r146343923) [](ancestors = 146343923)

Reply:
> Changed to tuple. However, I'm not sure the allocation was critical here, because it's an error scenario.

I believe tuple is also better from the API design perspective, we expect two things back, let's make it clear and enforceable.